File: src/base/SinglePageScraper.java
package base;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.io.IOException;

/**
 * Created by szamani on 12/18/2017.
 */
public abstract class SinglePageScraper extends PageScraper {
    protected String url;
    protected Document siteText;

    public SinglePageScraper(String country, String university, String department,
                             String url) {
        super(country, university, department);
        this.url = url;

        try {
            setSiteText();
        } catch (IOException e) {
            e.printStackTrace();
        }

        setDir();
        createJson();
        saveToFile();
    }

    protected void setSiteText() throws IOException {
        siteText = Jsoup.connect(url).get();
    }
}

File: beautifulsoup/base/SinglePageScraper.py
import requests
from bs4 import BeautifulSoup

from base.PageScraper import PageScraper


class SinglePageScraper(PageScraper):
    def __init__(self, url, university, department):
        super().__init__(university, department)
        self.url = url
        self.site_text = self.extract_html()
        self.soup = self.create_soup()
        self.create_json()
        self.save_to_file()
        self.save_exception()

    def extract_html(self):
        return requests.get(url=self.url).text

    def create_soup(self):
        return BeautifulSoup(self.site_text,
                             super().parser)
